internal/naming/labels.go

```diff
@@ -244,6 +247,7 @@ func PGBackRestDedicatedLabels(clusterName string) labels.Set {
 	commonLabels := PGBackRestLabels(clusterName)
 	operatorConfigLabels := map[string]string{
 		LabelPGBackRestDedicated: "",
+		LabelData:                DataPGBackRest,
```

Comment:
🔧 These labels also go into the repo host STS selector which is not allowed to change. When the operator deploys with this change, an existing cluster can no longer reconcile:

```
error="StatefulSet.apps \"mycluster-repo-host\" is invalid: spec: Forbidden: updates to statefulset spec for fields other than 'replicas', 'template', and 'updateStrategy' are forbidden"
```

Reply:
I updated the way this label is applied so that it is now set directly in `generateRepoHostIntent`. This should correct the selector issue as I understand it.